addons/mail/static/tests/discuss/core/public_web/discuss.test.js

```diff
@@ -1,86 +1,21 @@
-import { waitForChannels, waitUntilSubscribe } from "@bus/../tests/bus_test_helpers";
+import { waitUntilSubscribe } from "@bus/../tests/bus_test_helpers";
 import {
-    click,
     contains,
     defineMailModels,
     openDiscuss,
     start,
     startServer,
 } from "@mail/../tests/mail_test_helpers";
 import { describe, expect, test } from "@odoo/hoot";
-import { tick } from "@odoo/hoot-mock";
 import { EventBus } from "@odoo/owl";
-import { Command, patchWithCleanup, withUser, mockService } from "@web/../tests/web_test_helpers";
+import { Command, mockService, patchWithCleanup, withUser } from "@web/../tests/web_test_helpers";
 import { browser } from "@web/core/browser/browser";
 
 import { rpc } from "@web/core/network/rpc";
 
 describe.current.tags("desktop");
 defineMailModels();
 
-test("bus subscription updated when joining/leaving thread as non member", async () => {
-    const pyEnv = await startServer();
-    const johnUser = pyEnv["res.users"].create({ name: "John" });
-    const johnPartner = pyEnv["res.partner"].create({ name: "John", user_ids: [johnUser] });
-    const channelId = pyEnv["discuss.channel"].create({
-        channel_member_ids: [Command.create({ partner_id: johnPartner })],
-        name: "General",
-    });
-    await start();
-    await openDiscuss(channelId);
-    await waitForChannels([`discuss.channel_${channelId}`]);
-    await click("[title='Channel Actions']");
-    await click(".o-dropdown-item:contains('Leave Channel')");
-    await click("button", { text: "Leave Conversation" });
-    await waitForChannels([`discuss.channel_${channelId}`], { operation: "delete" });
-});
-
-test("bus subscription updated when joining locally pinned thread", async () => {
-    const pyEnv = await startServer();
-    const channelId = pyEnv["discuss.channel"].create({
-        channel_member_ids: [],
-        name: "General",
-    });
-    await start();
-    await openDiscuss(channelId);
-    await waitForChannels([`discuss.channel_${channelId}`]);
-    await click("[title='Invite People']");
-    await click(".o-discuss-ChannelInvitation-selectable", {
-        text: "Mitchell Admin",
-    });
-    await click(".o-discuss-ChannelInvitation [title='Invite']:enabled");
-    await waitForChannels([`discuss.channel_${channelId}`], { operation: "delete" });
-});
-
-test.skip("bus subscription kept after receiving a message as non member", async () => {
```

Comment:
Note: test removed. It's been skipped for a while and was introduced when the shadowing feature relied on the discuss side bar subscription (`isLocallyPinned`) which is not the case anymore. Even then, receiving a message is not linked to the subscription logic. With the test added in this PR, the coverage is good enough:
- Subscription updated when joining/leaving a channel.
- Subscription updated when pinning/unpinning locally.
- Subscription updated when opening/closing chat window of a thread the user isn't member of.